### PR TITLE
Only allow requests with correct host.

### PR DESCRIPTION
**Pull request by MancunianSam**
I've set this up so that the default behaviour of the load balancer listener is to send a 403 to any request.
I then add a rule which sends requests through to the target group if the header is set to the auth url.
This is working and it's stopped keycloak sending me a password reset link for my-very-evil-service.com.

### CODE REVIEW

File: alb/variables.tf [hunk 1]
```diff
@@ -78,3 +78,12 @@ variable "target_id" {
 variable "vpc_id" {
   description = "VPC ID"
 }
+
+variable "own_host_header_only" {
+  default = false
```
Comment (TomJKing) [inline alb/variables.tf:83]: Maybe a description for the variable would be helpful?

**Review by TomJKing — COMMENTED** [review]
Looks good. Just one minor suggestion

File: alb/variables.tf (hunk 1 above)
Comment (MancunianSam) [reply alb/variables.tf]: Yep, good point. I've added one in.

File: alb/variables.tf [hunk 2]
```diff
@@ -78,3 +78,13 @@ variable "target_id" {
 variable "vpc_id" {
   description = "VPC ID"
 }
+
+variable "own_host_header_only" {
+  description = "Only allow requests where the host header is set the the value of the host variable"
```
Comment (TomJKing) [inline alb/variables.tf:83]: Typo, should read: "... header is set **to the** value ..."

**Review by TomJKing — APPROVED** [review]

**Pull request merged**